pkg/auth/auth.go

```diff
@@ -0,0 +1,250 @@
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	auth = autorest.NewBearerAuthorizer(token)
+	return auth, err
+}
+
+// SetAuthConfig sets the configuration needed for Authentication
+func (c *Config) SetAuthConfig() error {
+	var err error
+	c.Cloud = cloud.AzurePublic
+
+	if authFilepath := os.Getenv("AZURE_AUTH_LOCATION"); authFilepath != "" {
+		auth := &Authentication{}
+		err = auth.NewAuthenticationFromFile(authFilepath)
+		if err != nil {
+			return err
+		}
+		c.AuthConfig = auth
+	}
+
+	if aksCredFilepath := os.Getenv("AKS_CREDENTIAL_LOCATION"); aksCredFilepath != "" {
+		c.AKSCredential, err = NewAKSCredential(aksCredFilepath)
+		if err != nil {
+			return err
+		}
+
+		if c.AKSCredential != nil {
+			var clientId string
+			if !strings.EqualFold(c.AKSCredential.ClientID, "msi") {
+				clientId = c.AKSCredential.ClientID
+			}
+
+			//Set Azure cloud environment
+			c.Cloud = getCloudConfiguration(c.AKSCredential.Cloud)
+			c.AuthConfig = NewAuthentication(
+				clientId,
+				c.AKSCredential.ClientSecret,
+				c.AKSCredential.SubscriptionID,
+				c.AKSCredential.TenantID,
+				c.AKSCredential.UserAssignedIdentityID)
+		}
+
+		if clientID := os.Getenv("AZURE_CLIENT_ID"); clientID != "" {
+			c.AuthConfig.ClientID = clientID
+		}
+
+		if clientSecret := os.Getenv("AZURE_CLIENT_SECRET"); clientSecret != "" {
+			c.AuthConfig.ClientSecret = clientSecret
+		}
+
+		if userIdentityClientId := os.Getenv("VIRTUALNODE_USER_IDENTITY_CLIENTID"); userIdentityClientId != "" {
+			c.AuthConfig.UserIdentityClientId = userIdentityClientId
+		}
+		c.AuthConfig.UseUserIdentity = (len(c.AuthConfig.ClientID) == 0)
+
+		if c.AuthConfig.UseUserIdentity {
+			if len(c.AuthConfig.UserIdentityClientId) == 0 {
+				return fmt.Errorf("neither AZURE_CLIENT_ID or VIRTUALNODE_USER_IDENTITY_CLIENTID is being set")
+			}
+
+			log.G(context.TODO()).Info("Using user identity for Authentication")
+		}
+
+		if tenantID := os.Getenv("AZURE_TENANT_ID"); tenantID != "" {
+			c.AuthConfig.TenantID = tenantID
+		}
+
+		if subscriptionID := os.Getenv("AZURE_SUBSCRIPTION_ID"); subscriptionID != "" {
+			c.AuthConfig.SubscriptionID = subscriptionID
+		}
+	}
+
+	resource := c.Cloud.Services[cloud.ResourceManager].Endpoint
+
+	c.Authorizer, err = c.getAuthorizer(resource)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Authentication represents the Authentication file for Azure.
+type Authentication struct {
+	ClientID             string `json:"clientId,omitempty"`
+	ClientSecret         string `json:"clientSecret,omitempty"`
+	SubscriptionID       string `json:"subscriptionId,omitempty"`
+	TenantID             string `json:"tenantId,omitempty"`
+	UseUserIdentity      bool   `json:"useUserIdentity,omitempty"`
+	UserIdentityClientId string `json:"userIdentityClientId,omitempty"`
+}
+
+// NewAuthenticationFromFile returns an Authentication struct from file path
+func (a *Authentication) NewAuthenticationFromFile(filepath string) error {
```

Comment:
If you don't plan to expose this method to be called in other modules, make it an internal method (newAuthenticationFromFile).

Reply:
done.